compose/ui/ui-uikit/src/uikitMain/objc/CMPUIKitUtils/CMPUIKitUtils/CMPMetalDrawablesHandler.h

```diff
@@ -0,0 +1,29 @@
+//
+//  CMPMetalDrawablesHandler.h
+//  CMPUIKitUtils
+//
+//  Created by Ilia.Semenov on 07/06/2024.
+//
+
+#import <Foundation/Foundation.h>
+#import <QuartzCore/QuartzCore.h>
+#import <Metal/Metal.h>
+
+#import "CMPMacros.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+@interface CMPMetalDrawablesHandler : NSObject
+
+- (instancetype)initWithMetalLayer:(CAMetalLayer *)metalLayer;
+
+- (void * CMP_OWNED)nextDrawable;
```

Comment:
Just thinking about slightly more generic solution.
What do you think about creating analog of "Manual Reference Counter" that can wrap various classes and hide them from Kotlin GC?
Or maybe even play a bit with NSProxy to provide sufficient interface for Kotlin.

Reply:
I tried to model this approach in my head, but didn't succeed. Do you have proof-of-concept available?